app/services/ui_notifications.rb

```diff
@@ -0,0 +1,31 @@
+# Namespace for Foreman UI notifications event handling
+module UiNotifications
+  class Base
+    attr_reader :state, :subject
+
+    def initialize(subject, state)
+      raise(Foreman::Exception, 'must provide notification subject') if subject.nil?
+      @subject = subject
+      raise(Foreman::Exception, 'invalid state') unless allowed_states.include?(state)
+      @state = state
+    end
+
+    def deliver!
+      raise(Foreman::Exception, 'not Implemented')
+    end
+
+    protected
+
+    def user
+      User.anonymous_api_admin
+    end
+
+    def allowed_states
```

Comment:
Please leave a comment explaining why `[]` is here, it's basically forcing the exception in the initializer